sshlauncher@sumo/files/sshlauncher@sumo/applet.js

```diff
@@ -106,31 +190,48 @@ MyApplet.prototype = {
     let menuitemEdit = new PopupMenu.PopupMenuItem(_("Edit SSH config"));
     menuitemEdit.connect('activate', Lang.bind(this, this.editConfig));
     this.menu.addMenuItem(menuitemEdit);
-    this.menu.addMenuItem(new PopupMenu.PopupSeparatorMenuItem());
-    let menuitemUpdate = new PopupMenu.PopupMenuItem(_("Force Update from SSH config"));
-    menuitemUpdate.connect('activate', Lang.bind(this, this.updateMenu));
-    this.menu.addMenuItem(menuitemUpdate);
   },
 
   connectTo: function(hostname) {
-    let flags = "";
-    if (this.sshHeadless) {
-      flags = " -fN ";
+    let command = "";
+    let terminal = "";
+
+    if (!this.empty(this._customCommand)) { // Custom command set in settings
+      command = this._customCommand;
+      terminal = command.split(" ")[0]; // Get terminal name from custom command
     }
-    if (this.sshForwardX) {
-      flags = " -X " + flags;
+    else {
+      let term = this.gsettings.get_string("exec");
+      if (this.empty(term)) {
+        this.sendNotification(_("SSH Launcher"), _("Error: I can't open the terminal.\nYou need to set a default terminal first in Preferred Applications!\n\n(More help in Right Click->About->More Info)"));
+        return;
+      }
+
+      terminal = term;
+      command = term + " ";
+      command += this.buildTermFlags(this.getTermOptions(term), hostname);
+
+      let termArg = this.gsettings.get_string("exec-arg");
+      if (!this.isExecArgCorrect(term, termArg)) {};
```

Comment:
What is the correct action when we know that the argument to run a command for the terminal is incorrect (obtained from gsettings)?
Some options:
* Notify about it in the notification and do nothing
* Notify about it in notification and override the argument to make sure the terminal is opened
* Do not use the gsettings exec-arg at all (or only as a fallback)?

Atm there is no feedback if the command has run successfully in script.